### ISSUE TITLE
Wake from sleep issue w RG300 IPS screen

### ISSUE THREAD

**Issue by killbankers** [retrofw/retrofw.github.io#16]
I've been playing around with RetroFW 2.0 for the last couple days and notice that if I let my RG300 with IPS screen go to sleep that it often will not wake up properly.  

Sometimes the screen just stays black (but backlight is on), sometimes it will come back on with varying levels of brightness (sometimes super dark/dim and other times with blown out blacks), and sometimes the screen comes back on like normal but lightly flickers.  These issues may resolve themselves after a minute or so but there are times I need to reboot the system to fix it.  Launching a game and quitting back to the GUI will often get the screen back to normal as well.

**Comment by alex-soderqvist** [comment]
I think I have similar issues. It flickers after being woken up, and sometime I can see vertical dark lines overlayed somehow. It is linked to edge cases like waking up, and also connecting/disconnecting the usb-c cable. I assumed it was a hardware issue, and accepted it due to the cheap nature of the device. But if it can be fixed in software I would be more than happy to test potential solutions. I also own an RG300 IPS.

**Comment by killbankers** [comment]
Yes, I also have seen it while connecting the USB cable as well.  Maybe has something to do with power management (sleep and USB would both affect system power)?  I hope/assume it can be fixed via software as it definitely wasn't there with RetroFW 1.X.  Thanks!

**Comment by malaccio** [comment]
i have the same issue, hold the sleep/power button for two second, it go to sleep but unable to wake up. Just a reboot solves the problem. Is this issue present also in v2.0?

**Comment by pingflood** [comment]
Yes. It's a known issue. We are working on it but no fix so far.

**Comment by malaccio** [comment]
Now on RG300, using 2.2 version for now I am able to wake up but once I wake up from suspend mode the colors are too faded.. it seems that luminosity is increased from original color up to 200%

**Comment by panic2k** [comment]
On RetroFW 2.0 sleep / resume on RG300 IPS works for me without any issues if Min-CPU in Settings is at least set to 230MHZ.

**Comment by malaccio** [comment]
Is it yours a real IPS? I mean 2.8"? Which firmware version did you install?

Il Lun 16 Nov 2020, 13:59 panic2k <notifications@github.com> ha scritto:

**Comment by malaccio** [comment]
> On RetroFW 2.0 sleep / resume on RG300 IPS works for me without any issues if Min-CPU in Settings is at least set to 230MHZ.

But how do you set the minimum cpu usage in settings?
If I remember well there was a setting in pymenu, but for retrofw in general or Gmenu2x? 
Thanks

**Comment by Darku87** [comment]
> > On RetroFW 2.0 sleep / resume on RG300 IPS works for me without any issues if Min-CPU in Settings is at least set to 230MHZ.
> 
> But how do you set the minimum cpu usage in settings?
> If I remember well there was a setting in pymenu, but for retrofw in general or Gmenu2x?
> Thanks

I can't find that setting either. Do they refer to the clock speed option on each app/emulator/game link?

**Comment by panic2k** [comment]
In Gmenu2x this setting could be found in:
Settings -> CPU Settings -> Minimum CPU clock.
The default setting was 342.
This setting is only used in standby mode as i've verified in ssh session. Max CPU defines the maximum clock which could be set for an application and default clock defines clock speed in Gmenu2x without anything running. The clock speed in every app/emulator settings is set before emulator is started.

As i got my RG300 IPS i've directly messed around with settings. I wondered why device begun starts to hang after standby. It's just a idea what it could be. Maybe there is another bug but i've not run into other issues with similar impacts. As i've written before i'm on RetroFW 2.0, and it has (maybe) nothing to do with the standby issues on 2.2.

**Comment by Darku87** [comment]
> In Gmenu2x this setting could be found in:
> Settings -> CPU Settings -> Minimum CPU clock.
> The default setting was 342.
> This setting is only used in standby mode as i've verified in ssh session. Max CPU defines the maximum clock which could be set for an application and default clock defines clock speed in Gmenu2x without anything running. The clock speed in every app/emulator settings is set before emulator is started.
> 
> As i got my RG300 IPS i've directly messed around with settings. I wondered why device begun starts to hang after standby. It's just a idea what it could be. Maybe there is another bug but i've not run into other issues with similar impacts. As i've written before i'm on RetroFW 2.0, and it has (maybe) nothing to do with the standby issues on 2.2.

That's strange, I dont hace the CPU settings option in settings... Can you please share the content of your gmenu2x.conf ?

**Comment by panic2k** [comment]
for shure:

skin="Skin-xfce"
wallpaper="skins/Skin-xfce/wallpapers/xubuntu-quantal.png"
lang="German"
usbMode="Ask"
batteryType="BL-5B"
backlight=100
backlightTimeout=120
linkCols=1
cpuMax=684
globalVolume=50
link=19
cpuMin=230
sectionBar=1
section=2
cpuMenu=570
linkRows=6
powerTimeout=30
linkLabel=0
outputLogs=1

it's running fine on my device without any issues.

**Comment by malaccio** [comment]
Mmmh, do you have retrofw 2.2?
In my gmenu2x.conf I only have:
Skin
Lang
Wallpaper
TuMode
SkinFont
Backlight
BacklightTimeout
GlobalVolume
PowerTimeout




Il Mer 2 Dic 2020, 23:39 panic2k <notifications@github.com> ha scritto:

**Comment by panic2k** [comment]
i'm on 2.0, tried 2.2 today to reproduce, but couldn't get it to boot with another sd, i just got white flashing screens with 5B and 3B images. but i've seen that gmenunx is used, the settings should be the same and by default at 342. don't know if other power management is implemented, but i hope the devs keep an eye on freqs since i can reproduce similar behaviour with retrofw 2.0. maybe i try later if i have more spare time. For shure you could try to add 'cpuMin=342' to gmenu2x.conf meanwhile

**Comment by Darku87** [comment]
Tried that config but didn't solve the issue :(
